plugin/reload/reload.go

```diff
@@ -46,6 +49,14 @@ func (r *reload) interval() time.Duration {
 	return r.dur
 }
 
+func parse(corefile caddy.Input) ([]byte, error) {
+	serverBlocks, err := caddyfile.Parse(corefile.Path(), bytes.NewReader(corefile.Body()), nil)
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(serverBlocks)
```

Comment:
Are we sure this is deterministic? That is, are we sure it always results in the same ordering? Otherwise we could end up with continuously reloading. We may need the MD5 to navigate through any map structure, sorting the map keys and creating a canonical string representation of the whole file.

Reply:
From my understanding `json.Marshal` is deterministic, although the documentation does a poor job at describing this. There is a part of the Marshal docs that states `map keys are sorted and used as JSON object keys` (https://github.com/golang/go/blob/5fae09b7386de26db59a1184f62fc7b22ec7667b/src/encoding/json/encode.go#L135). Looking at the code it appears map keys and struct fields are sorted going back to at least Go v1.4.

Information added to README about the behavior with version context.